services/addons/images/firmware_manager/firmware_manager.py

```diff
@@ -213,6 +225,18 @@ def firmware_upgrade_completed():
                     ),
                     500,
                 )
+        else:
+            increment_consecutive_failure_count_for_rsu(request_args["rsu_ip"])
+            if is_rsu_at_max_retries_limit(request_args["rsu_ip"]):
+                logging.error(f"RSU {request_args['rsu_ip']} has reached the maximum number of upgrade retries. Setting target_firmware_version to firmware_version and resetting consecutive failures count.")

```

Comment:
question(non-blocking): does this log mean we're updating the target_firmware_version to the current version (like the version already installed on the rsu) or to the firmware_version we tried but failed to install (the newest firmware version)? I'm pretty sure it's meant to be the former, but for some reason my brain is having a hard time feeling certain about what this log means 😅

Reply:
The `rsus` table contains two columns related to firmware: `firmware_version` and `target_firmware_version`. The `firmware_version` column shows the current firmware version installed on the RSU, while the `target_firmware_version` indicates the desired firmware version we want to install.

By setting `target_firmware_version` to match the current value of the `firmware_version` column, we are instructing the firmware manager to keep the RSU on its existing firmware and prevent any upgrades.